airflow/models/taskmixin.py

```diff
@@ -112,6 +115,31 @@ def __rlshift__(self, other: DependencyMixin | Sequence[DependencyMixin]):
         self.__rshift__(other)
         return self
 
+    def set_setup_teardown_ctx_dependencies(self, other: DependencyMixin | Sequence[DependencyMixin]):
+        if not SetupTeardownContext.active:
+            return
+        from airflow.models.xcom_arg import PlainXComArg
+
+        op1 = self
+        if isinstance(self, PlainXComArg):
+            op1 = self.operator
+        if getattr(op1, "is_setup") or getattr(op1, "is_teardown"):
+            return
+        SetupTeardownContext.update_context_map(op1)
```

Comment:
We should add annotations to `update_context_map` as well. I have a feeling this is also not handled well inside there—the argument name is `operator` but there are obviously cases where `op1` and `other` is not an operator in this function.